packages/core-snapshots/lib/db/index.js

```diff
@@ -8,20 +8,25 @@ const { rawQuery } = require('./utils')
 const columns = require('./utils/column-set')
 
 module.exports = class Database {
-  constructor (db, pgp) {
-    if (pgp) {
-      this.db = db
-      this.pgp = pgp
+  constructor (database) {
+    if (database) {
+      this.db = database.db
+      this.pgp = database.pgp
+      this.__createColumnSets()
+      logger.info('Snapshots: reusing core-database-postgres connection from running core')
+
       return
     }
+
     try {
       const pgp = require('pg-promise')({ promiseLib: promise })
       this.pgp = pgp
       this.db = pgp(container.resolveOptions('database').connection)
       this.__createColumnSets()
+      logger.info('Snapshots: Database connected')
     } catch (error) {
       logger.error(`Error while connecting to postgres: ${error}`)
-      process.exit(1)
+      throw new Error(error.stack)
```

Comment:
This should be a `process.exit(1)` to 100% make sure the application actually exits or set `process. exitCode = 1` before you throw the error.